### ISSUE TITLE
Unable to Create Droplet from Java Main Application

### ISSUE THREAD

**Issue by nebhale** [cloudfoundry-incubator/lattice#161]
Currently, when attempting to create a droplet from a [Java Main Application](https://github.com/cloudfoundry/java-test-applications/tree/master/java-main-application), the attempt fails.  The problem appears to lie somewhere in the handling of the archive:

```bash
wlan-10-81-130-157 (HEAD detached at v0.3.0): ltc build-droplet java-main-application https://github.com/cloudfoundry/java-buildpack.git -p /Users/bhale/dev/sources/java-test-applications/java-main-application/build/libs/java-main-application-1.0.0.BUILD-SNAPSHOT.jar | tee ~/Desktop/output.txt
Submitted build of java-main-application
[BUILD|0] Creating container
[BUILD|0] Successfully created container
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/spring-common-1.0.0.BUILD-SNAPSHOT.jar, /tmp/app/lib/spring-common-1.0.0.BUILD-SNAPSHOT.jar.zip or /tmp/app/lib/spring-common-1.0.0.BUILD-SNAPSHOT.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/core-1.0.0.BUILD-SNAPSHOT.jar, /tmp/app/lib/core-1.0.0.BUILD-SNAPSHOT.jar.zip or /tmp/app/lib/core-1.0.0.BUILD-SNAPSHOT.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/spring-boot-starter-web-1.2.4.RELEASE.jar, /tmp/app/lib/spring-boot-starter-web-1.2.4.RELEASE.jar.zip or /tmp/app/lib/spring-boot-starter-web-1.2.4.RELEASE.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/spring-rabbit-1.4.5.RELEASE.jar, /tmp/app/lib/spring-rabbit-1.4.5.RELEASE.jar.zip or /tmp/app/lib/spring-rabbit-1.4.5.RELEASE.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/spring-data-mongodb-1.7.1.RELEASE.jar, /tmp/app/lib/spring-data-mongodb-1.7.1.RELEASE.jar.zip or /tmp/app/lib/spring-data-mongodb-1.7.1.RELEASE.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/spring-data-redis-1.5.1.RELEASE.jar, /tmp/app/lib/spring-data-redis-1.5.1.RELEASE.jar.zip or /tmp/app/lib/spring-data-redis-1.5.1.RELEASE.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/spring-context-4.1.7.RELEASE.jar, /tmp/app/lib/spring-context-4.1.7.RELEASE.jar.zip or /tmp/app/lib/spring-context-4.1.7.RELEASE.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/spring-core-4.1.7.RELEASE.jar, /tmp/app/lib/spring-core-4.1.7.RELEASE.jar.zip or /tmp/app/lib/spring-core-4.1.7.RELEASE.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/spring-jdbc-4.1.7.RELEASE.jar, /tmp/app/lib/spring-jdbc-4.1.7.RELEASE.jar.zip or /tmp/app/lib/spring-jdbc-4.1.7.RELEASE.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/h2-1.4.187.jar, /tmp/app/lib/h2-1.4.187.jar.zip or /tmp/app/lib/h2-1.4.187.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/jedis-2.6.3.jar, /tmp/app/lib/jedis-2.6.3.jar.zip or /tmp/app/lib/jedis-2.6.3.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/spring-boot-starter-1.2.4.RELEASE.jar, /tmp/app/lib/spring-boot-starter-1.2.4.RELEASE.jar.zip or /tmp/app/lib/spring-boot-starter-1.2.4.RELEASE.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/jackson-databind-2.4.6.jar, /tmp/app/lib/jackson-databind-2.4.6.jar.zip or /tmp/app/lib/jackson-databind-2.4.6.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/hibernate-validator-5.1.3.Final.jar, /tmp/app/lib/hibernate-validator-5.1.3.Final.jar.zip or /tmp/app/lib/hibernate-validator-5.1.3.Final.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/spring-web-4.1.6.RELEASE.jar, /tmp/app/lib/spring-web-4.1.6.RELEASE.jar.zip or /tmp/app/lib/spring-web-4.1.6.RELEASE.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/spring-webmvc-4.1.6.RELEASE.jar, /tmp/app/lib/spring-webmvc-4.1.6.RELEASE.jar.zip or /tmp/app/lib/spring-webmvc-4.1.6.RELEASE.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/amqp-client-3.5.1.jar, /tmp/app/lib/amqp-client-3.5.1.jar.zip or /tmp/app/lib/amqp-client-3.5.1.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/spring-retry-1.1.2.RELEASE.jar, /tmp/app/lib/spring-retry-1.1.2.RELEASE.jar.zip or /tmp/app/lib/spring-retry-1.1.2.RELEASE.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/spring-amqp-1.4.5.RELEASE.jar, /tmp/app/lib/spring-amqp-1.4.5.RELEASE.jar.zip or /tmp/app/lib/spring-amqp-1.4.5.RELEASE.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/spring-messaging-4.1.6.RELEASE.jar, /tmp/app/lib/spring-messaging-4.1.6.RELEASE.jar.zip or /tmp/app/lib/spring-messaging-4.1.6.RELEASE.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/spring-data-commons-1.10.1.RELEASE.jar, /tmp/app/lib/spring-data-commons-1.10.1.RELEASE.jar.zip or /tmp/app/lib/spring-data-commons-1.10.1.RELEASE.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/mongo-java-driver-2.13.0.jar, /tmp/app/lib/mongo-java-driver-2.13.0.jar.zip or /tmp/app/lib/mongo-java-driver-2.13.0.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/spring-context-support-4.0.9.RELEASE.jar, /tmp/app/lib/spring-context-support-4.0.9.RELEASE.jar.zip or /tmp/app/lib/spring-context-support-4.0.9.RELEASE.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/commons-pool2-2.3.jar, /tmp/app/lib/commons-pool2-2.3.jar.zip or /tmp/app/lib/commons-pool2-2.3.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/spring-boot-1.2.4.RELEASE.jar, /tmp/app/lib/spring-boot-1.2.4.RELEASE.jar.zip or /tmp/app/lib/spring-boot-1.2.4.RELEASE.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/spring-boot-autoconfigure-1.2.4.RELEASE.jar, /tmp/app/lib/spring-boot-autoconfigure-1.2.4.RELEASE.jar.zip or /tmp/app/lib/spring-boot-autoconfigure-1.2.4.RELEASE.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/spring-boot-starter-logging-1.2.4.RELEASE.jar, /tmp/app/lib/spring-boot-starter-logging-1.2.4.RELEASE.jar.zip or /tmp/app/lib/spring-boot-starter-logging-1.2.4.RELEASE.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/snakeyaml-1.14.jar, /tmp/app/lib/snakeyaml-1.14.jar.zip or /tmp/app/lib/snakeyaml-1.14.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/jackson-annotations-2.4.0.jar, /tmp/app/lib/jackson-annotations-2.4.0.jar.zip or /tmp/app/lib/jackson-annotations-2.4.0.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/jackson-core-2.4.6.jar, /tmp/app/lib/jackson-core-2.4.6.jar.zip or /tmp/app/lib/jackson-core-2.4.6.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/validation-api-1.1.0.Final.jar, /tmp/app/lib/validation-api-1.1.0.Final.jar.zip or /tmp/app/lib/validation-api-1.1.0.Final.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/jboss-logging-3.1.3.GA.jar, /tmp/app/lib/jboss-logging-3.1.3.GA.jar.zip or /tmp/app/lib/jboss-logging-3.1.3.GA.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/classmate-1.0.0.jar, /tmp/app/lib/classmate-1.0.0.jar.zip or /tmp/app/lib/classmate-1.0.0.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/jul-to-slf4j-1.7.12.jar, /tmp/app/lib/jul-to-slf4j-1.7.12.jar.zip or /tmp/app/lib/jul-to-slf4j-1.7.12.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/log4j-over-slf4j-1.7.12.jar, /tmp/app/lib/log4j-over-slf4j-1.7.12.jar.zip or /tmp/app/lib/log4j-over-slf4j-1.7.12.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/logback-classic-1.1.3.jar, /tmp/app/lib/logback-classic-1.1.3.jar.zip or /tmp/app/lib/logback-classic-1.1.3.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/logback-core-1.1.3.jar, /tmp/app/lib/logback-core-1.1.3.jar.zip or /tmp/app/lib/logback-core-1.1.3.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/commons-logging-1.2.jar, /tmp/app/lib/commons-logging-1.2.jar.zip or /tmp/app/lib/commons-logging-1.2.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/spring-tx-4.1.7.RELEASE.jar, /tmp/app/lib/spring-tx-4.1.7.RELEASE.jar.zip or /tmp/app/lib/spring-tx-4.1.7.RELEASE.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/spring-beans-4.1.7.RELEASE.jar, /tmp/app/lib/spring-beans-4.1.7.RELEASE.jar.zip or /tmp/app/lib/spring-beans-4.1.7.RELEASE.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/spring-aop-4.1.7.RELEASE.jar, /tmp/app/lib/spring-aop-4.1.7.RELEASE.jar.zip or /tmp/app/lib/spring-aop-4.1.7.RELEASE.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/aopalliance-1.0.jar, /tmp/app/lib/aopalliance-1.0.jar.zip or /tmp/app/lib/aopalliance-1.0.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/spring-expression-4.1.7.RELEASE.jar, /tmp/app/lib/spring-expression-4.1.7.RELEASE.jar.zip or /tmp/app/lib/spring-expression-4.1.7.RELEASE.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/jcl-over-slf4j-1.7.12.jar, /tmp/app/lib/jcl-over-slf4j-1.7.12.jar.zip or /tmp/app/lib/jcl-over-slf4j-1.7.12.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/slf4j-api-1.7.12.jar, /tmp/app/lib/slf4j-api-1.7.12.jar.zip or /tmp/app/lib/slf4j-api-1.7.12.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/spring-boot-starter-tomcat-1.2.4.RELEASE.jar, /tmp/app/lib/spring-boot-starter-tomcat-1.2.4.RELEASE.jar.zip or /tmp/app/lib/spring-boot-starter-tomcat-1.2.4.RELEASE.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/tomcat-embed-core-8.0.23.jar, /tmp/app/lib/tomcat-embed-core-8.0.23.jar.zip or /tmp/app/lib/tomcat-embed-core-8.0.23.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/tomcat-embed-el-8.0.23.jar, /tmp/app/lib/tomcat-embed-el-8.0.23.jar.zip or /tmp/app/lib/tomcat-embed-el-8.0.23.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/tomcat-embed-logging-juli-8.0.23.jar, /tmp/app/lib/tomcat-embed-logging-juli-8.0.23.jar.zip or /tmp/app/lib/tomcat-embed-logging-juli-8.0.23.jar.ZIP.
[BUILD|0] unzip:  cannot find or open /tmp/app/lib/tomcat-embed-websocket-8.0.23.jar, /tmp/app/lib/tomcat-embed-websocket-8.0.23.jar.zip or /tmp/app/lib/tomcat-embed-websocket-8.0.23.jar.ZIP.
[BUILD|0] -----> Java Buildpack Version: ec1a5c9 | https://github.com/cloudfoundry/java-buildpack.git#ec1a5c9
[BUILD|0] [Buildpack]                      ERROR Compile failed with exception #<RuntimeError: No container can run this application>
[BUILD|0] No container can run this application
[BUILD|0] Failed to compile droplet: exit status 1
[BUILD|0] Exit status 1
Build failed: Exited with status 1
```

Note that the [Web Application](https://github.com/cloudfoundry/java-test-applications/tree/master/web-application) seems to work properly.

**Comment by davidwadden** [comment]
@nebhale - we were able to reproduce this today.  the problem looks to be with how gradle is packaging the jar:

by adding the following to `droplet_runner.go`, 
```
			&models.RunAction{
				Path: "/usr/bin/find",
				Dir:  "/",
				Args: []string{"/tmp/app", "-ls"},
				User: "root",
			},
```

we learned that this JAR was being created with Windows-style file permissions:

```
±  |master ✓| →  zipinfo -l ~/workspace/buildpacks/java-test-applications/java-main-application/build/libs/java-main-application-1.0.0.BUILD-SNAPSHOT.jar
Archive:  /Users/pivotal/workspace/buildpacks/java-test-applications/java-main-application/build/libs/java-main-application-1.0.0.BUILD-SNAPSHOT.jar   19264785   119
-rw----     2.0 fat        0 bX        2 defN 10-Aug-15 10:07 META-INF/
-rw----     2.0 fat      189 bl      148 defN 10-Aug-15 10:07 META-INF/MANIFEST.MF
-rw----     2.0 fat        0 bl        2 defN 10-Aug-15 10:07 lib/
-rw----     1.0 fat     5888 b-     5888 stor 10-Aug-15 10:07 lib/spring-common-1.0.0.BUILD-SNAPSHOT.jar
```

which leads to directories not having the executable bit set when they're unpacked in the container:
```
08/10 11:02:04.72 [BUILD|0]  24025    0 drw-r--r--   1 vcap     vcap           34 Aug 10 16:02 /tmp/app
08/10 11:02:04.72 [BUILD|0]  24026    0 drw-r--r--   1 vcap     vcap           22 Aug 10 10:07 /tmp/app/META-INF
08/10 11:02:04.72 [BUILD|0]  24027    4 -rw-r--r--   1 vcap     vcap          189 Aug 10 10:07 /tmp/app/META-INF/MANIFEST.MF
...
```

thus resulting in the error messages "unzip: cannot find or open..."

by unzipping and re-zipping the JAR on our Mac, 

```
○ → jar xf ~/workspace/buildpacks/java-test-applications/java-main-application/build/libs/java-main-application-1.0.0.BUILD-SNAPSHOT.jar
○ → zip -r ../my-jar.zip *
  adding: META-INF/ (stored 0%)
  adding: META-INF/MANIFEST.MF (deflated 22%)
  adding: com/ (stored 0%)
  ...
○ → zipinfo ../my-jar.zip
Archive:  ../my-jar.zip   17291823 bytes   119 files
drwxr-xr-x  3.0 unx        0 bx stor 10-Aug-15 10:07 META-INF/
-rw-r--r--  3.0 unx      189 tx defN 10-Aug-15 10:07 META-INF/MANIFEST.MF
drwxr-xr-x  3.0 unx        0 bx stor 10-Aug-15 10:07 com/
```

then resuming the build + launch droplet flow,
```
○ → ltc build-droplet java-main-application https://github.com/cloudfoundry/java-buildpack.git -p ~/workspace/buildpacks/my-jar.zip
Submitted build of java-main-application
08/10 11:02:03.67 [BUILD|0] Successfully created container
08/10 11:02:04.14 [BUILD|0] ABOUT TO DL BITS
08/10 11:02:04.15 [BUILD|0] Exit status 0
08/10 11:02:04.70 [BUILD|0] DOWNLOADED BITS
08/10 11:02:04.70 [BUILD|0] Exit status 0
08/10 11:02:04.72 [BUILD|0]  24025    0 drwxr-xr-x   1 vcap     vcap           34 Aug 10 16:02 /tmp/app
08/10 11:02:04.72 [BUILD|0]  24026    0 drwxr-xr-x   1 vcap     vcap           22 Aug 10 10:07 /tmp/app/META-INF
08/10 11:02:04.72 [BUILD|0]  24027    4 -rw-r--r--   1 vcap     vcap          189 Aug 10 10:07 /tmp/app/META-INF/MANIFEST.MF
08/10 11:02:04.72 [BUILD|0]  24028    0 drwxr-xr-x   1 vcap     vcap           18 Aug 10 10:07 /tmp/app/com
...
08/10 11:02:04.77 [BUILD|0] RUNNING BUILDER
08/10 11:02:04.77 [BUILD|0] Exit status 0
08/10 11:02:07.54 [BUILD|0] -----> Java Buildpack Version: ec1a5c9 | https://github.com/cloudfoundry/java-buildpack.git#ec1a5c9
08/10 11:02:13.60 [BUILD|0] -----> Downloading Open Jdk JRE 1.8.0_51 from https://download.run.pivotal.io/openjdk/trusty/x86_64/openjdk-1.8.0_51.tar.gz (5.6s)
08/10 11:02:14.71 [BUILD|0]        Expanding Open Jdk JRE to .java-buildpack/open_jdk_jre (1.1s)
08/10 11:02:17.52 [BUILD|0] -----> Downloading Open JDK Like Memory Calculator 1.1.1_RELEASE from https://download.run.pivotal.io/memory-calculator/trusty/x86_64/memory-calculator-1.1.1_RELEASE (2.8s)
08/10 11:02:17.53 [BUILD|0]        Memory Settings: -Xmx54613K -Xms54613K -XX:MaxMetaspaceSize=64M -XX:MetaspaceSize=64M -Xss568K
08/10 11:02:19.07 [BUILD|0] -----> Downloading Spring Auto Reconfiguration 1.8.0_RELEASE from https://download.run.pivotal.io/auto-reconfiguration/auto-reconfiguration-1.8.0_RELEASE.jar (1.5s)
08/10 11:02:28.01 [BUILD|0] Exit status 0
08/10 11:02:28.03 [BUILD|0] RAN BUILDER
08/10 11:02:28.03 [BUILD|0] Exit status 0
08/10 11:02:28.45 [BUILD|0] Uploaded /tmp/droplet to http://192.168.11.11.xip.io:8444/blobs/java-main-application/droplet.tgz.
08/10 11:02:28.46 [BUILD|0] Exit status 0
08/10 11:02:28.49 [BUILD|0] Uploaded /tmp/result.json to http://192.168.11.11.xip.io:8444/blobs/java-main-application/result.json.
08/10 11:02:28.49 [BUILD|0] Exit status 0
08/10 11:02:28.53 [BUILD|0] Deleted http://192.168.11.11.xip.io:8444/blobs/java-main-application/bits.zip.
08/10 11:02:28.53 [BUILD|0] Exit status 0
Build completed
○ → ltc launch-droplet java-main-droplet java-main-application -m 512
No port specified. Defaulting to 8080.
Creating App: java-main-droplet
.08/10 11:03:09.05 [APP|0] Creating container
08/10 11:03:09.35 [APP|0] Successfully created container
....08/10 11:03:13.44 [HEALTH|0] healthcheck failed
08/10 11:03:13.45 [HEALTH|0] Exit status 1
08/10 11:03:13.97 [HEALTH|0] healthcheck failed
08/10 11:03:13.98 [HEALTH|0] Exit status 1
08/10 11:03:14.05 [APP|0]   .   ____          _            __ _ _
08/10 11:03:14.05 [APP|0]  /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
08/10 11:03:14.05 [APP|0] ( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
08/10 11:03:14.05 [APP|0]  \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
08/10 11:03:14.05 [APP|0]   '  |____| .__|_| |_|_| |_\__, | / / / /
08/10 11:03:14.05 [APP|0]  =========|_|==============|___/=/_/_/_/
08/10 11:03:14.05 [APP|0]  :: Spring Boot ::        (v1.2.4.RELEASE)
...
08/10 11:03:19.85 [APP|0] 2015-08-10 16:03:19.854  INFO 21 --- [           main] o.s.j.e.a.AnnotationMBeanExporter        : Registering beans for JMX exposure on startup
08/10 11:03:19.86 [APP|0] 2015-08-10 16:03:19.864  INFO 21 --- [           main] o.s.c.support.DefaultLifecycleProcessor  : Starting beans in phase 2147483647
08/10 11:03:19.93 [APP|0] 2015-08-10 16:03:19.934  INFO 21 --- [           main] s.b.c.e.t.TomcatEmbeddedServletContainer : Tomcat started on port(s): 8080 (http)
08/10 11:03:19.93 [APP|0] 2015-08-10 16:03:19.936  INFO 21 --- [           main] c.g.c.test.ApplicationConfiguration      : Started ApplicationConfiguration in 6.325 seconds (JVM running for 7.023)
08/10 11:03:20.04 [HEALTH|0] healthcheck passed
08/10 11:03:20.06 [HEALTH|0] Exit status 0

java-main-droplet is now running.
App is reachable at:
http://java-main-droplet.192.168.11.11.xip.io
```

then, this app works.

have you tried running this cloudfoundry backed by diego ?  we're interested to know if that works before deciding how to address this issue.

@davidwadden / @crawsible

**Comment by nebhale** [comment]
I haven't tried running it against a Diego-backed Cloud Foundry.  I'm not setup to do it at the moment, but am happy to try it for you (can you point me at some instructions for that please?).  The ZIP behavior from Gradle (specifically), isn't likely to change for the better any time soon.  This odd behavior is well-known in the Java community and people work around it on the run side.

**Comment by onsi** [comment]
@nebhale follow these instructions on PWS: https://github.com/cloudfoundry-incubator/diego-design-notes/blob/master/migrating-to-diego.md#targeting-diego

+ @ematpl - if we have issues with the Java buildpack Diego needs to know asap.

**Comment by nebhale** [comment]
@onsi I'm checking right now.

**Comment by nebhale** [comment]
`java-main-application` works as expected against PWS with Diego.  I'll verify the rest of the application types as well.

**Comment by nebhale** [comment]
I've gone back and run all of the [`java-test-application`s](https://github.com/cloudfoundry/java-test-applications) against Lattice v0.3.0.  The following projects failed:

* `dist-zip-application`
* ``
* ``
* ``

**Comment by nebhale** [comment]
I've gone back and run all of the [`java-test-application`s](https://github.com/cloudfoundry/java-test-applications) against Lattice v0.3.0.  The following projects failed:

* `dist-zip-application`
* `java-main-application`
* `spring-boot-cli-jar-application`

I'd bet they all suffer from the directory problem.

**Comment by nebhale** [comment]
@onsi I've just verified all 10 test application types on PWS/Diego and all work as expected.

**Comment by ematpl** [comment]
Thanks for the check against Diego on PWS, @nebhale!

Best,
Eric

**Comment by davidwadden** [comment]
Hi @nebhale,

This issue has been addressed in the story listed above, [[#100854344](https://www.pivotaltracker.com/story/show/100854344)]. It'll be included in Lattice v0.3.1, which will be released this week.

Closing this ticket for now; as always if you think we've missed something, please feel free to re-open.

--
@davidwadden && @menicosia
